backend/app/services/IngestStorage.scala

```diff
@@ -63,19 +65,51 @@ class S3IngestStorage private(client: S3Client, bucket: String) extends IngestSt
 
   override def delete(key: (Long, UUID)) = {
     Either.catchNonFatal{
-      client.aws.deleteObject(bucket, dataKey(key))
-      client.aws.deleteObject(bucket, metadataKey(key))
+      client.aws.deleteObject(ingestBucket, dataKey(key))
+      client.aws.deleteObject(ingestBucket, metadataKey(key))
+    }.leftMap(UnknownFailure.apply)
+  }
+
+  override def sendToDeadLetterBucket(key: Key): Either[Failure, Unit] = {
+    Either.catchNonFatal {
+      client.aws.copyObject(ingestBucket, dataKey(key), deadLetterBucket, dataKey(key))
+      client.aws.copyObject(ingestBucket, metadataKey(key), deadLetterBucket, metadataKey(key))
+    } match {
+      // if copy succeeded, delete the files from the ingest bucket
+      case Right(_) => delete(key)
+      case Left(failure) => Left(UnknownFailure(failure))
+    }
+  }
+
+  override def retryDeadLetters(): Either[UnknownFailure, Unit] = {
+    Either.catchNonFatal {
+      val result = client.aws.listObjects(deadLetterBucket)
+      if (!result.isTruncated) {
+        logger.info(s"Sending ${result.getObjectSummaries.size()} files from dead letter bucket to ingest bucket.")
+        result.getObjectSummaries.forEach{ summary =>
+          Try(client.aws.copyObject(deadLetterBucket, summary.getKey, ingestBucket, summary.getKey))
+            // on success, clean up the file from the dead letter bucket
+            .map(_ => client.aws.deleteObject(deadLetterBucket, summary.getKey))
+        }
+      } else {
+        val msg = "Too many dead letter files to resync via API. Try using aws cli e.g aws s3 sync s3://deadletterbucket s3://ingestbucket"
+        logger.error(msg)
+        throw new Error(msg)
+      }
     }.leftMap(UnknownFailure.apply)
   }
```

Comment:
I know this task should be very quick. But just wondering, do we not care if the metadata is copied before the actual file? Because if the file is copied first, and if it quickly be picked up by the worker before the metadata is copied, the worker may fail

Reply:
Ohh good point, I had thought about this a little but just hoped s3 would be fast 🤦🏻 

I've done a bit more research - it looks like files are ingested only when the 'data' bit has been uploaded ([getNextBatch](https://github.com/guardian/giant/blob/main/backend/app/ingestion/phase2/IngestStorePolling.scala#L166) calls [list](https://github.com/guardian/giant/blob/4f61f5543df91057e9dec6c91e9ca7de966d040a/backend/app/services/IngestStorage.scala#L36) which only fetches the data prefix) 

So I've refactored this to do the metadata events first, which should mean that the reingestion only starts once the data bit has finished copying

Reply:
(changes are in https://github.com/guardian/giant/pull/149/commits/5a144231655409b4bf0861cfb7115e1a7e345003, I'll test on playground before merging)

Reply:
Amazing, really like the usage of the partition 👌